### ISSUE TITLE
Multiple sessions to a camera

### ISSUE THREAD

**Issue by jelockwood** [AlexxIT/go2rtc#835]
It appears many cameras have a limit on how many simultaneous sessions the camera can directly support. I believe Doorbird for example only officially support just one session. This I suspect is partly down to short sightedness and also a terrible trend for products to be increasingly designed by bean counters and not engineers. (That is designed down to the lowest possible cost with a resulting sacrifice of capabilities, these days even food is increasingly designed by accountants rather than chefs.)

Whilst this problem is not strictly speaking a go2rtc issue it could affect how one uses or configures go2rtc and I am seeking advice related to this.

What potentially makes this issue worse is that Apple HomeKIt Video can require two simultaneous sessions. This is because Apple added a new feature to allow a popup notification to appear on HomeKit clients e.g. iPhone or AppleTV with a box showing a preview of the video stream. If this is showing on the Apple TV and you answer on an iPhone you can get a message saying all available streams are in use. See the following discussion.

https://forums.macrumors.com/threads/too-few-continuous-video-streams-on-robin-proline-doorbell.2333834/

As will be clear from the above discussion, the Robin Proline HomeKit models only support two streams. I do however get a slight impression the Robin Proline SIP models might support three streams. Can the Robin Proline SIP models be used as a source for go2rtc? Will it support two-way audio? It maybe then that getting the SIP model, feeding that in to go2rtc and having go2rtc exporting it to HomeKIt would be better than the Proline HomeKIt model.

If go2rtc is acting both as a HomeKit proxy for the HomeKit version of the Proline and outputting an RTSP stream does that count as using up one of the HomeKit streams making this issue worse or because it is being proxied it is still available leaving two HomeKit streams available for use? Is there any way to allow more HomeKit streams, perhaps by having the camera added to HASS and then exported from HASS so that HASS acts as the 'proxy'?

Note: The issue of multiple sessions would presumably also affect being able to access the camera simultaneously via an RTSP client and feeding to an NVR.

**Comment by AlexxIT** [comment]
> Can the Robin Proline SIP models be used as a source for go2rtc?

I don't know. Never heard about this devices. go2rtc doesn't support SIP proto and I don't have any experience with it.

> If go2rtc is acting both as a HomeKit proxy for the HomeKit version of the Proline and outputting an RTSP stream does that count as using up one of the HomeKit streams

Yes. go2rtc takes one stream for homekit source (input). Also go2rtc unlimited count of streams as homekit output

**Comment by jelockwood** [comment]
> Yes. go2rtc takes one stream for homekit source (input). Also go2rtc unlimited count of streams as homekit output

@AlexxIT Thank you, that was what I was hoping that by using go2rtc as an intermediary for HomeKit it would enable unlimited streams. Would I be right however in thinking this would have to be as a non-proxy config so that go2rtc rebuilds HomeKit streams rather than passing them through unaltered?

**Comment by AlexxIT** [comment]
Yes. You can create first stream for HomeKit source. Then create second stream as RTSP link to first stream. And use second stream as HomeKit server. So it won't be HomeKit proxy mode.
